src/socket.rs

```diff
@@ -138,15 +141,16 @@ mod linux {
     }
 }
 
+#[cfg(all(target_os = "linux"))]
 const EINVAL: i32 = 22;
 pub async fn relay(
     downstream: &mut tokio::net::TcpStream,
     upstream: &mut tokio::net::TcpStream,
-    zero_copy_enabled: bool,
+    _zero_copy_enabled: bool,
```

Comment:
underscore seems odd given it *is* used. Maybe we cna make the entire function differ based on OS to avoid?